src/renderer/components/+preferences/preferences.tsx

```diff
@@ -22,249 +22,105 @@
 import "./preferences.scss";
 
 import React from "react";
-import moment from "moment-timezone";
-import { computed, observable, reaction, makeObservable } from "mobx";
+import { observable, reaction, makeObservable } from "mobx";
 import { disposeOnUnmount, observer } from "mobx-react";
 
-import { isWindows } from "../../../common/vars";
-import { AppPreferenceRegistry, RegisteredAppPreference } from "../../../extensions/registries/app-preference-registry";
-import { UserStore } from "../../../common/user-store";
-import { ThemeStore } from "../../theme.store";
-import { Input } from "../input";
+import { AppPreferenceKind, AppPreferenceKindRegistry, AppPreferenceRegistry, RegisteredAppPreference } from "../../../extensions/registries/app-preference-registry";
 import { SubTitle } from "../layout/sub-title";
-import { Select, SelectOption } from "../select";
-import { HelmCharts } from "./helm-charts";
-import { KubectlBinaries } from "./kubectl-binaries";
 import { navigation } from "../../navigation";
 import { Tab, Tabs } from "../tabs";
-import { FormSwitch, Switcher } from "../switch";
-import { KubeconfigSyncs } from "./kubeconfig-syncs";
 import { SettingLayout } from "../layout/setting-layout";
-
-enum Pages {
-  Application = "application",
-  Proxy = "proxy",
-  Kubernetes = "kubernetes",
-  Telemetry = "telemetry",
-  Extensions = "extensions",
-  Other = "other"
-}
+import { iter } from "../../utils";
 
 @observer
 export class Preferences extends React.Component {
-  @observable httpProxy = UserStore.getInstance().httpsProxy || "";
-  @observable shell = UserStore.getInstance().shell || "";
-  @observable activeTab = Pages.Application;
+  @observable activeTab: string = AppPreferenceKind.Application;
 
   constructor(props: {}) {
     super(props);
     makeObservable(this);
   }
 
-  @computed get themeOptions(): SelectOption<string>[] {
-    return ThemeStore.getInstance().themes.map(theme => ({
-      label: theme.name,
-      value: theme.id,
-    }));
-  }
-
-  timezoneOptions: SelectOption<string>[] = moment.tz.names().map(zone => ({
-    label: zone,
-    value: zone,
-  }));
-
   componentDidMount() {
     disposeOnUnmount(this, [
       reaction(() => navigation.location.hash, hash => {
         const fragment = hash.slice(1); // hash is /^(#\w.)?$/
 
-        if (fragment) {
-          // ignore empty framents
-          document.getElementById(fragment)?.scrollIntoView();
+        if (fragment.length > 0) {
+          const settingsItem = AppPreferenceRegistry.getInstance().getItems().find(item => item.id === fragment);
+
+          if (settingsItem) {
+            this.activeTab = settingsItem.showInPreferencesTab;
+            setTimeout(() => {
+              document.getElementById(fragment)?.scrollIntoView();
+            }, 150);
+          }
```

Comment:
And the moving out of the in-repo settings to the registry was done because this is the correct logic (the previous logic no longer worked because the id wasn't necessarily in the DOM.

Technically, it would be more correct to do this scrollIntoView after the next render but I was not able to find a succinct method of doing that.